crypto/asn1/a_time.c

```diff
@@ -471,13 +471,15 @@ static const char _asn1_mon[12][4] = {
 };
 
 /* returns 1 on success, 0 on BIO write error or parse failure */
-int ASN1_TIME_print(BIO *bp, const ASN1_TIME *tm)
+int ASN1_TIME_print(BIO *bp, const ASN1_TIME *tm, int date_format)
```

Comment:
You will need to add an ASN1_TIME_print_ex function.. Since this is an existing public API.
ASN1_TIME_print() should just call ASN1_TIME_print_ex().

Reply:
This seems to have revealed a misunderstanding I had about these functions, and perhaps why this was broken before.

I've added a new function ASN1_TIME_print_ex function, but it's no longer recognized from any of the application code. I get the following linker error when I try to reference it, even though it's declared in `asn1.h`

```
/usr/bin/ld: apps/openssl-bin-ca.o: in function `do_body.isra.0.constprop.0':
ca.c:(.text+0x17d3): undefined reference to `ASN1_TIME_print_ex'
/usr/bin/ld: apps/openssl-bin-crl.o: in function `crl_main':
crl.c:(.text+0x74d): undefined reference to `ASN1_TIME_print_ex'
/usr/bin/ld: crl.c:(.text+0x79b): undefined reference to `ASN1_TIME_print_ex'
/usr/bin/ld: apps/openssl-bin-x509.o: in function `x509_main':
x509.c:(.text+0x2143): undefined reference to `ASN1_TIME_print_ex'
collect2: error: ld returned 1 exit status
```

Would you happen to know how to resolve this?

Reply:
The trick is to run `make update` (only possible on a Unix like environment, sorry).  The crucial file here is `util/libcrypto.num`, which is generated by `make update` and which contains the info on what symbols should be exported in the shared library